src/Binary/BinaryInterface.php

```diff
@@ -13,18 +13,9 @@
 
 interface BinaryInterface
 {
-    /**
-     * @return string
-     */
-    public function getContent();
+    public function getContent(): string;
 
-    /**
-     * @return string
-     */
-    public function getMimeType();
+    public function getMimeType(): ?string;
```

Comment:
I've left this as nullable because this code:

https://github.com/liip/LiipImagineBundle/blob/8a57c556227cc9854ac7334ae28a3c752c8fef3d/Tests/Imagine/Data/DataManagerTest.php#L174

and 

https://github.com/liip/LiipImagineBundle/blob/8a57c556227cc9854ac7334ae28a3c752c8fef3d/Imagine/Data/DataManager.php#L131-L143

In the last case, we can move the exception right after:

`$mimeType = $this->mimeTypeGuesser->guess($binary);`

I'll add an extra commit doing this and then we decide what to do

Reply:
ah, i was wondering why that is nullable.

i think its better to have an exception than having to check the return value for null. then if it is acceptable that the mime type is unknown, the exception can be caught, but the user of the interface does not stumble accidentally over the null. i wonder if we should have a specific exception for this, instead of the generic LogicException. it is not really a "logic" problem imho.

if we change this, we need to add a note to the upgrade guide, i think.

Reply:
I think I'll open an issue about this and leave it as nullable for now so we can finish the static typing.